Makefile

```diff
@@ -1,85 +1,16 @@
-include Makefile_header.mk
-location = $(CURDIR)/$(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
-WHERE := $(location)
-$(info ** -> $(WHERE))
-$(info ** ------------------------------------------------------------------ **)
-NVCC := $(shell command -v nvcc 2> /dev/null)
-
-SHELL := /bin/bash # force avoidance of dash as shell
-#
-# Build specific config
-#
-CONFIG=make/config.mk
-include $(CONFIG)
-
-VERSION=make/version.mk
-include $(VERSION)
-
-MAJOR_MINOR=$(shell echo $(BASE_VERSION) | sed 's/.*\(^[0-9][0-9]*\.[0-9][0-9]*\).*/\1/g' )
-
-# System specific stuff
-include src/config2.mk
-
-ifdef NVCC
-ifeq ($(shell test $(CUDA_MAJOR) -ge 9; echo $$?),0)
-    $(warning Compiling with Cuda9 or higher)
-    XGB_CUDA ?= -DGPU_COMPUTE_VER="35;52;60;61;70"
-else
-    $(warning Compiling with Cuda8 or lower)
-    # >=52 required for kmeans for larger data of size rows/32>2^16
-    XGB_CUDA ?= -DGPU_COMPUTE_VER="35;52;60;61"
-endif
-endif
-
-# Location of local directory with dependencies
-DEPS_DIR = deps
-
-# Detect OS
-OS := $(shell uname)
-## Python has crazy ideas about os names
-ifeq ($(OS), Darwin)
-		PY_OS ?= "macosx"
-else
-		PY_OS ?= $(OS)
-endif
-
-# see if have ccache for faster compile times if no changes to file
-theccache=$(shell echo `which ccache`)
-ifeq ($(theccache),)
-		theccacheclean=
-else
-		theccacheclean=$(theccache) -C
-endif
-
-RANDOM := $(shell bash -c 'echo $$RANDOM')
-LOGEXT=$(RANDOM)$(shell date +'_%Y.%m.%d-%H:%M:%S')
-NUMPROCS := $(shell cat /proc/cpuinfo|grep processor|wc -l)
-
-#
-# Docker image tagging
-#
-DOCKER_VERSION_TAG ?= "latest"
-
-#
-# Setup S3 access credentials
-#
-S3_CMD_LINE := aws s3
-
-#
-# BUILD_INFO setup
-#
-H2O4GPU_COMMIT ?= $(shell git rev-parse HEAD)
-H2O4GPU_BUILD_DATE := $(shell date)
-H2O4GPU_BUILD ?= "LOCAL BUILD @ $(shell git rev-parse --short HEAD) build at $(H2O4GPU_BUILD_DATE)"
-H2O4GPU_SUFFIX ?= "+local_$(shell git describe --always --dirty)"
+include make/Makefile_header.mk
```

Comment:
Great improvements in terms of removing dead code and simplifying things. I think another good exercise is to review the Makefile and see if we totally understand each target. How much are repeating? Can anything be simplified. I think you have taken a huge step in this approach. Just something for others to keep in mind going forward.